src/Patcher.cpp

```diff
@@ -562,17 +562,17 @@ void* FixPtr(const void *pointer, HMODULE module) {
       reinterpret_cast<uintptr_t>(module) +
       reinterpret_cast<IMAGE_DOS_HEADER*>(module)->e_lfanew)->OptionalHeader;
 
-    if (optionalHeader->Magic == IMAGE_NT_OPTIONAL_HDR32_MAGIC) {
-      preferredAddress = optionalHeader->ImageBase;
-    }
-    else if (optionalHeader->Magic == IMAGE_NT_OPTIONAL_HDR64_MAGIC) {
-      preferredAddress = reinterpret_cast<IMAGE_OPTIONAL_HEADER64*>(optionalHeader)
-                         ->ImageBase;
-    }
-    else {
+    if ((optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR32_MAGIC) && 
+       ((optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR64_MAGIC)) {
       // Not a valid PE image
       return nullptr;
-    }
+		}
+
+    #ifdef _WIN64
+      preferredAddress = reinterpret_cast<IMAGE_OPTIONAL_HEADER64*>(optionalHeader)->ImageBase;
+    #elif

```

Comment:
I believe this should be #else. On MSVC, a compiler error of E0127, expected a statement is given.

```suggestion
    #else
```

Reply:
Ahh yes, thanks for catching that. I was translating from the original `if` - `else if` - `else` structure, and didn't update it.

Neat little suggest mechanism.